### DISCUSSION TITLE
Typings of a wrapper component over Controller

### DISCUSSION THREAD

**Discussion by aslupik** [react-hook-form/react-hook-form#1707] [General] [answered]
I'm trying to write a HOC that wraps the Controller component. Its props would simply be the props of the Controller component, and it would forward them to the Controller component. Intended usage would be a drop-in replacement for `Controller`:

`<Controller as={SomeComponent} name={'..'} propOfSomeComponent={...} />`
with
`<HOC  as={SomeComponent} name={'..'} propOfSomeComponent={...} />`

I'm having an awfully hard time coming up with the type of the props though. I'm not sure I even fully understand the type definition in controller.d.ts. Could you lend a hand?

**Comment by bluebill1049** [comment, marked as answer]
@kotarella1110 could you please give a hand on type definition on Controller. 

Side note: what's the reason to write a wrapper on top of another?

**Reply by kotarella1110** [reply]
How is this?

```ts
import React from 'react';
import { Controller } from 'react-hook-form';

type HOCProps = React.ComponentProps<typeof Controller>;
```

**Reply by aslupik** [reply]
You're a true wizard! I wish I had known of this type before.

**Reply by aslupik** [reply]
@bluebill1049 the wrapper would compute the value of some properties so that it doesn't need to be done repeatedly everywhere.

**Reply by winnie-development** [reply]
This is exactly the question and answer I needed, thanks!
